### PR TITLE
fix pytest strings (#438)

### PR DESCRIPTION
**Pull request by rupertford**

### CODE REVIEW

**Comment by rupertford** [comment]
Ready for first review from @arporter subject to travis being happy.

**Comment by rupertford** [comment]
Coverage is supposedly not 100% but I've only changed the tests.

File: src/psyclone/tests/dynamo0p3_multigrid_test.py [hunk 1]
```diff
@@ -104,9 +104,9 @@ def test_invalid_mesh_type():
     name = "restrict_kernel_type"
     with pytest.raises(ParseError) as excinfo:
         _ = DynKernMetadata(ast, name=name)
-    print(str(excinfo))
+    print(str(excinfo.value))
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/psyclone/tests/dynamo0p3_multigrid_test.py]: DOne.

File: src/psyclone/tests/dynamo0p3_multigrid_test.py [hunk 2]
```diff
@@ -118,9 +118,9 @@ def test_invalid_mesh_specifier():
...
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/psyclone/tests/dynamo0p3_multigrid_test.py]: Done.

File: src/psyclone/tests/f2pygen_test.py [hunk 3]
```diff
@@ -1395,8 +1397,9 @@ def test_modulegen_add_wrong_parent():
     sub = SubroutineGen(module_wrong, name="testsubroutine")
     with pytest.raises(RuntimeError) as err:
         module.add(sub)
-    print(str(err))
-    assert "because it is not a descendant of it or of any of" in str(err)
+    print(str(err.value))
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/psyclone/tests/f2pygen_test.py]: Done.

**Comment by rupertford** [comment]
Ready for next review from @arporter subject to travis being happy. Regarding the test that was not failing, well spotted: it was passing because the string we were comparing with was also the name of the exception. I've updated the test. I've also updated setup.py, again well spotted.

**Comment by arporter** [comment]
All requested changes have been made and pip installation works OK (I tried `pip install .[test]` in a clean virtualenv). Please resolve the conflicts with master and then I'll merge.

**Comment by rupertford** [comment]
Resolved conflicts and checked through the code.

**Pull request merged**